src/main/main.js

```diff
@@ -201,19 +141,101 @@ class Main {
         json: true
       });
       token = deployment.token;
-      console.log('Using new token...');
       await fs.writeFile(tokenPath, token, { encoding: 'utf-8' }, function(
         err,
         data
       ) {
         if (err) {
           console.log(err);
+          this.mainWindow.webContents.send('pub-error', err.message);
         }
       });
     }
-    console.log('token: ' + token);
     return token;
   }
+
+  executeOnProjectOpen(file) {
+    this.filePath = file;
+
+    this.workingDir = getWorkingDirectory(p, this.filePath);
+
+    // Instantiate an Idyll instance
+    this.idyll = Idyll({
+      inputFile: this.filePath,
+      output: this.workingDir + '/build/',
+      componentFolder: this.workingDir + '/components/',
+      dataFolder: this.workingDir + '/data',
+      layout: 'centered',
+      theme: 'default'
+    });
+
+    // filter to catch all requests to static folder
+    const staticContentFilter = { urls: ['static/*'] };
+    this.mainWindow.webContents.session.webRequest.onBeforeRequest(
+      staticContentFilter,
+      (details, callback) => {
+        const { url } = details;
+        if (url.indexOf(`${this.electronWorkingDir}/static/`) > -1) {
+          const localURL = url.replace(
+            this.electronWorkingDir,
+            this.workingDir
+          );
+          callback({
+            cancel: false,
+            redirectURL: encodeURI(localURL)
+          });
+        } else {
+          callback({
+            cancel: false
+          });
+        }
+      }
+    );
+
+    // Serialize file contents
+    let fileContent = '';
+    try {
+      fileContent = fs.readFileSync(file).toString();
+    } catch (err) {
+      console.log(err);
+    }
+
+    // Compile contents
+    compile(fileContent, {})
+      .then(ast => {
+        // Get project URL if it exists
+        const tokenPath = getTokenPath(p, this.workingDir);
+
+        let url = '';
+        try {
+          const token = fs.readFileSync(tokenPath, { encoding: 'utf-8' });
+          const tokenUrl = this.store.getTokenUrlByToken(token);
+          if (tokenUrl) {
+            url = tokenUrl.url;
+          } else {
+            // TODO: Get URL from idyll pub api here
```

Comment:
Is this currently a blocker?

Reply:
Once the idyll-pub server has that GET endpoint it should be an easy add-in I think. But for right now, I just commented where that logic should be!